### ISSUE TITLE
More values available for Deye-SUN-2000G3-EU (deye_4mppt)

### ISSUE THREAD

**Issue by bolausson** [catalinsanda/org.openhab.binding.solarman#10]
Hej,

thanks for this great binding.

There are more values which can directly be gathered from the Deye-SUN-2000G3-EU (deye_4mppt).
I found an [deye_4mppt yaml definition](https://github.com/jedie/inverter-connect/pull/68/commits/8c7ac22d8e0f62164b320ca661529741c6abf822) which works pretty well with a minior flaw which I reported [here](https://github.com/jedie/inverter-connect/issues/78).

Example of the available values in the mentioned yaml file:
![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/10600678/d99fdbaf-76c9-4b2e-95d1-413ccac3ddec)

Patched file attached:
[deye_4mppt.yaml.txt](https://github.com/catalinsanda/org.openhab.binding.solarman/files/12171991/deye_4mppt.yaml.txt)

Would be great if you could integrate those additional values into your binding.

Greetings,
Bjoern

**Comment by catalinsanda** [comment]
Sure, will have a look, thanks!

**Comment by bolausson** [comment]
Thanks a lot!
I am not very good at Java, else I would have tried to contribute some code :(

**Comment by catalinsanda** [comment]
Try the new release and let me know how it works. I haven't tested the values, as I don't have the right inverter.

https://github.com/catalinsanda/org.openhab.binding.solarman/releases/tag/v0.2.2

**Comment by bolausson** [comment]
Hej!
Thanks a lot, it kind of works!

I updated the binding (Stopped OpenHAB, copied 0.2.2 jar to /usr/share/openhab/addons/, deleted 0.2.1 jar, started OpenHAB)
I was able to see the addional values but no longer the brevious values.
Interestingly, when I klick on "save" when on the "channel" page, I suddenly see the "old channels", if I klick "save" again, I see the "new channels", if I klick another time on "save", I can see the "old channels"... But I never get to see all channel (from 0.2.1 and the additional channels from 0.2.2) at once.

2.2.1 channels:
![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/10600678/cbc0fffc-0561-4922-92f1-16a05802d1e7)

2.2.2 channels (all channels from 2.2.1 missing):
![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/10600678/57d8877b-3315-45cc-96bf-a7e524e006de)

The only channel that doesn't seem to work is "Update Time".

It looks like the gathered values depend on "what channels I currently see". So either the 2.2.1 channels are logged or the 2.2.2 channels, but I never see all channels being updated (in Grafana or events.log)

```
root@OpenHAB $ grep solarman:logger openhab.log
2023-07-30 13:28:32.542 [WARN ] [core.thing.internal.ThingManagerImpl] - Channel types or config descriptions for thing 'solarman:logger:4acf902443' are missing in the respective registry for more than 120s. In case it does not happen immediately after an upgrade, it should be fixed in the binding.
2023-07-30 13:28:32.546 [WARN ] [core.thing.internal.ThingManagerImpl] - Failed to normalize configuration for thing 'solarman:logger:4acf902443': {thing/channel=Type description solarman:inverter_id for solarman:logger:4acf902443:inverter#inverter_id not found, although we checked the presence before.}
```

Here is what I see in events.log (after I klicked "save" when I see the 2.2.2 channels in the UI, after that I am presented withthe 2.2.1 channels):
```
2023-07-30 13:44:54.229 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Startup_Selfchecking_Time_' changed from 0 to 60
2023-07-30 13:44:54.232 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit' changed from 52 Hz to 0 Hz
2023-07-30 13:44:54.241 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Rated_Power' changed from 2000 W to 0 W
2023-07-30 13:44:54.244 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_OverfrequencyLoadshedding_Enable' changed from OFF to ON
2023-07-30 13:44:54.247 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Starting_Point' changed from 50.2 Hz to 0 Hz
2023-07-30 13:44:54.250 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_ONOFF_Enable' changed from OFF to ON
2023-07-30 13:44:54.253 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Rated_Power' changed from 0 W to 2000 W
2023-07-30 13:44:54.257 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Island_Protection_Enable' changed from 0 to 1
2023-07-30 13:44:54.266 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Lower_Limit' changed from 0 V to 180 V
2023-07-30 13:44:54.269 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Active_Power_Regulations' changed from ON to OFF
2023-07-30 13:44:54.271 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Upp_Limit' changed from 0 V to 275 V
2023-07-30 13:44:54.274 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit' changed from 47.5 Hz to 0 Hz
2023-07-30 13:44:54.277 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit' changed from 0 Hz to 52 Hz
2023-07-30 13:44:54.281 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Startup_Selfchecking_Time_' changed from 60 to 0
2023-07-30 13:44:54.283 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Percentage' changed from 0 to 44
2023-07-30 13:44:54.286 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Percentage' changed from 44 to 0
2023-07-30 13:44:54.291 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_ONOFF_Enable' changed from ON to OFF
2023-07-30 13:44:54.297 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_OverfrequencyLoadshedding_Enable' changed from ON to OFF
2023-07-30 13:44:54.302 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_1' changed from 0.1 kWh to 0 kWh
2023-07-30 13:44:54.305 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Starting_Point' changed from 0 Hz to 50.2 Hz
2023-07-30 13:44:54.306 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_2' changed from 119.2 kWh to 0 kWh
2023-07-30 13:44:54.308 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit' changed from 0 Hz to 47.5 Hz
2023-07-30 13:44:54.309 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Lower_Limit' changed from 180 V to 0 V
2023-07-30 13:44:54.314 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_2' changed from 0.1 kWh to 0 kWh
2023-07-30 13:44:54.317 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Island_Protection_Enable' changed from 1 to 0
2023-07-30 13:44:54.322 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_1' changed from 0 kWh to 0.1 kWh
2023-07-30 13:44:54.324 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Upp_Limit' changed from 275 V to 0 V
2023-07-30 13:44:54.326 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_1' changed from 0 kWh to 117.1 kWh
2023-07-30 13:44:54.328 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Active_Power_Regulations' changed from OFF to ON
2023-07-30 13:44:54.330 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_2' changed from 0 kWh to 119.2 kWh
2023-07-30 13:44:54.331 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_1' changed from 117.1 kWh to 0 kWh
2023-07-30 13:44:54.334 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_2' changed from 0 kWh to 0.1 kWh


2023-07-30 13:45:54.237 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Startup_Selfchecking_Time_' changed from 0 to 60
2023-07-30 13:45:54.247 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit' changed from 52 Hz to 0 Hz
2023-07-30 13:45:54.254 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Rated_Power' changed from 2000 W to 0 W
2023-07-30 13:45:54.260 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_OverfrequencyLoadshedding_Enable' changed from OFF to ON
2023-07-30 13:45:54.262 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Starting_Point' changed from 50.2 Hz to 0 Hz
2023-07-30 13:45:54.274 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_ONOFF_Enable' changed from OFF to ON
2023-07-30 13:45:54.282 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Rated_Power' changed from 0 W to 2000 W
2023-07-30 13:45:54.291 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Island_Protection_Enable' changed from 0 to 1
2023-07-30 13:45:54.301 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Lower_Limit' changed from 0 V to 180 V
2023-07-30 13:45:54.303 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Active_Power_Regulations' changed from ON to OFF
2023-07-30 13:45:54.307 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Upp_Limit' changed from 0 V to 275 V
2023-07-30 13:45:54.307 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit' changed from 47.5 Hz to 0 Hz
2023-07-30 13:45:54.311 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit' changed from 0 Hz to 52 Hz
2023-07-30 13:45:54.314 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Startup_Selfchecking_Time_' changed from 60 to 0
2023-07-30 13:45:54.317 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Percentage' changed from 0 to 44
2023-07-30 13:45:54.320 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Percentage' changed from 44 to 0
2023-07-30 13:45:54.322 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_ONOFF_Enable' changed from ON to OFF
2023-07-30 13:45:54.324 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_OverfrequencyLoadshedding_Enable' changed from ON to OFF
2023-07-30 13:45:54.326 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_1' changed from 0.1 kWh to 0 kWh
2023-07-30 13:45:54.331 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Starting_Point' changed from 0 Hz to 50.2 Hz
2023-07-30 13:45:54.332 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_2' changed from 119.2 kWh to 0 kWh
2023-07-30 13:45:54.334 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit' changed from 0 Hz to 47.5 Hz
2023-07-30 13:45:54.335 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Lower_Limit' changed from 180 V to 0 V
2023-07-30 13:45:54.338 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_2' changed from 0.1 kWh to 0 kWh
2023-07-30 13:45:54.342 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Island_Protection_Enable' changed from 1 to 0
2023-07-30 13:45:54.345 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_1' changed from 0 kWh to 0.1 kWh
2023-07-30 13:45:54.349 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Voltage_Upp_Limit' changed from 275 V to 0 V
2023-07-30 13:45:54.350 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_1' changed from 0 kWh to 117.1 kWh
2023-07-30 13:45:54.352 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Active_Power_Regulations' changed from OFF to ON
2023-07-30 13:45:54.354 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_2' changed from 0 kWh to 119.2 kWh
2023-07-30 13:45:54.356 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production_1' changed from 117.1 kWh to 0 kWh
2023-07-30 13:45:54.359 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production_2' changed from 0 kWh to 0.1 kWh
```
You see, none of the additional 2.2.2 values.

Now if I click "save" again whilst I see the 2.2.1 channels in the UI, I get the following in the events.log:
```
2023-07-30 13:51:56.357 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from XXXXXXX to 0
2023-07-30 13:51:56.361 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Current' changed from 1.7 A to 0 A
2023-07-30 13:51:56.364 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 0 V to 232 V
2023-07-30 13:51:56.366 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 232 V to 0 V
2023-07-30 13:51:56.371 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 0 to 2
2023-07-30 13:51:56.375 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0 A to 0.4 A
2023-07-30 13:51:56.379 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 2 to 0
2023-07-30 13:51:56.383 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 29.2 V to 0 V
2023-07-30 13:51:56.385 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 0 V to 29.2 V
2023-07-30 13:51:56.387 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 0 °C to 31.8 °C
2023-07-30 13:51:56.388 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from 0 to XXXXXXX
2023-07-30 13:51:56.391 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_AC_Output_Power_Active' changed from 100 W to 0 W
2023-07-30 13:51:56.392 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 249.3 kWh to 0 kWh
2023-07-30 13:51:56.393 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 31.8 °C to 0 °C
2023-07-30 13:51:56.394 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 28.9 V to 0 V
2023-07-30 13:51:56.395 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0.4 A to 0 A
2023-07-30 13:51:56.397 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 0 kWh to 249.3 kWh
2023-07-30 13:51:56.398 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 0 A to 1.7 A
2023-07-30 13:51:56.399 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0.3 kWh to 0 kWh
2023-07-30 13:51:56.400 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 0 Hz to 50 Hz
2023-07-30 13:51:56.401 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0 kWh to 0.3 kWh
2023-07-30 13:51:56.402 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 50 Hz to 0 Hz
2023-07-30 13:51:56.404 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 1.7 A to 0 A
2023-07-30 13:51:56.409 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 0 V to 28.9 V


2023-07-30 13:52:56.366 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_AC_Output_Power_Active' changed from 0 W to 100 W
2023-07-30 13:52:56.370 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Current' changed from 0 A to 1.7 A
2023-07-30 13:52:56.377 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from XXXXXXX to 0
2023-07-30 13:52:56.384 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Current' changed from 1.7 A to 0 A
2023-07-30 13:52:56.387 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 0 V to 232 V
2023-07-30 13:52:56.391 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 232 V to 0 V
2023-07-30 13:52:56.394 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 0 to 2
2023-07-30 13:52:56.398 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0 A to 0.4 A
2023-07-30 13:52:56.409 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 2 to 0
2023-07-30 13:52:56.413 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 29.2 V to 0 V
2023-07-30 13:52:56.417 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 0 V to 29.2 V
2023-07-30 13:52:56.420 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 0 °C to 31.8 °C
2023-07-30 13:52:56.422 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from 0 to XXXXXXX 
2023-07-30 13:52:56.426 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_AC_Output_Power_Active' changed from 100 W to 0 W
2023-07-30 13:52:56.431 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 249.3 kWh to 0 kWh
2023-07-30 13:52:56.432 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 31.8 °C to 0 °C
2023-07-30 13:52:56.435 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 28.9 V to 0 V
2023-07-30 13:52:56.437 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0.4 A to 0 A
2023-07-30 13:52:56.439 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 0 kWh to 249.3 kWh
2023-07-30 13:52:56.441 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 0 A to 1.7 A
2023-07-30 13:52:56.443 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0.3 kWh to 0 kWh
2023-07-30 13:52:56.445 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 0 Hz to 50 Hz
2023-07-30 13:52:56.447 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0 kWh to 0.3 kWh
2023-07-30 13:52:56.449 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 50 Hz to 0 Hz
2023-07-30 13:52:56.451 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 1.7 A to 0 A
2023-07-30 13:52:56.456 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 0 V to 28.9 V
2023-07-30 13:53:56.345 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_AC_Output_Power_Active' changed from 0 W to 64 W
2023-07-30 13:53:56.353 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Current' changed from 0 A to 0.9 A
2023-07-30 13:53:56.368 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from XXXXXXX to 0
2023-07-30 13:53:56.380 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Current' changed from 0.9 A to 0 A
2023-07-30 13:53:56.386 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 0 V to 233 V
2023-07-30 13:53:56.392 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Voltage' changed from 233 V to 0 V
2023-07-30 13:53:56.397 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 0 to 2
2023-07-30 13:53:56.400 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0 A to 0.2 A
2023-07-30 13:53:56.404 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Running_Status' changed from 2 to 0
2023-07-30 13:53:56.406 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 29.2 V to 0 V
2023-07-30 13:53:56.409 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Voltage' changed from 0 V to 29.6 V
2023-07-30 13:53:56.412 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 0 °C to 30.1 °C
2023-07-30 13:53:56.415 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Inverter_ID' changed from 0 to XXXXXXX 
2023-07-30 13:53:56.418 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_AC_Output_Power_Active' changed from 64 W to 0 W
2023-07-30 13:53:56.421 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 249.3 kWh to 0 kWh
2023-07-30 13:53:56.423 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Radiator_Temperature' changed from 30.1 °C to 0 °C
2023-07-30 13:53:56.426 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 28.9 V to 0 V
2023-07-30 13:53:56.429 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Grid_Current' changed from 0.2 A to 0 A
2023-07-30 13:53:56.432 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Total_Production' changed from 0 kWh to 249.4 kWh
2023-07-30 13:53:56.435 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 0 A to 1 A
2023-07-30 13:53:56.438 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0.3 kWh to 0 kWh
2023-07-30 13:53:56.441 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 0 Hz to 50 Hz
2023-07-30 13:53:56.444 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_Daily_Production' changed from 0 kWh to 0.3 kWh
2023-07-30 13:53:56.446 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' changed from 50 Hz to 0 Hz
2023-07-30 13:53:56.448 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV2_Current' changed from 1 A to 0 A
2023-07-30 13:53:56.451 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'DeyeSUN2000G3EU_PV1_Voltage' changed from 0 V to 35.1 V
```
You see, none of the previous 2.2.1 channels. 


In general, the additional metrics work (if they are gatheres) - I guess I screwed it up the way I updated your binding.
Is there a "proper way" of installing and updating a binding which is not yet available in the "Community Marketplace"?

I will try to uninstall the binding and install 2.2.2 and re-link all the items - maybe it I see all channels after that.

Cheers,
Bjoern

**Comment by catalinsanda** [comment]
The update procedure is correct. That's exactly what I do myself.

This binding is a bit special, in the sense that channels are not defined statically in the `thing-types.xml` file like for the vast majority of bindings. Instead the channels are created dynamically when the binding is initialized based on the inverter type, to support all the inverters I have definitions for. There are probably other ways to skin this cat, probably using bridges and static types with static channels, which are created at build time from the inverter definitions, but I'm not sure when and if I'll be able to invest the time to look into this alternative.

In my own setup I define my items in an item file (https://www.openhab.org/docs/configuration/items.html), see an example in https://github.com/catalinsanda/org.openhab.binding.solarman/blob/main/README.md, and this has worked without issues so far. 

I'll try to setup items in the UI as well, maybe I can find what's the problem. My fear is that maybe some parts of OpenHAB are caching the channels and when the binding has updated them, it's started to misbehave.  I'll investigate and report back.

In the meantime I suggest you go with an items file as well. It's simple to put together and it's a lot easier to migrate from one version of OpenHAB to another.

**Comment by bolausson** [comment]
Thanks for the explanation!

I tried to remove an add the binding but it didn't help.

I will try to add an item file. I'll keep you posted!

Greetings,
Bjoern

**Comment by bolausson** [comment]
Okay, gave it a quick try by "exporting" the items I added via the UI to a tiems file and then import theis items file via the UI using the Developer-Tools "Add Items from Textual Definition". I was hoping to "update" the existing items but I get an error when klick "Add":

`An error occurred while creating the links and metadata: Bad Request`

Although I see in the event.lot:
```
2023-07-30 20:58:00.299 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_AC_Output_Frequency' has been updated.
2023-07-30 20:58:00.302 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_AC_Voltage' has been updated.
2023-07-30 20:58:00.305 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_Grid_Current' has been updated.
2023-07-30 20:58:00.308 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit' has been updated.
2023-07-30 20:58:00.310 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit' has been updated.
[...]
2023-07-30 20:58:00.469 [INFO ] [openhab.event.ItemUpdatedEvent      ] - Item 'DeyeSUN2000G3EU_Total_Production_4' has been updated.
```

Since it is getting dark now, I'll check tomorrow if this helped or now.

How do you "import" your items file?



Just for reference if someone wants to do the same - the items.file for all the channels:
```
Number:Frequency DeyeSUN2000G3EU_AC_Output_Frequency "AC Output Frequency" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#ac_output_frequency" }
Number:ElectricPotential DeyeSUN2000G3EU_AC_Voltage "AC Voltage" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#ac_voltage" }
Number:ElectricCurrent DeyeSUN2000G3EU_Grid_Current "Grid Current" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#grid_current" }
Number:Frequency DeyeSUN2000G3EU_Grid_Frequency_Lower_Limit "Grid Frequency Lower Limit" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#grid_frequency_lower_limit" }
Number:Frequency DeyeSUN2000G3EU_Grid_Frequency_Upper_Limit "Grid Frequency Upper Limit" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#grid_frequency_upper_limit" }
Number:ElectricPotential DeyeSUN2000G3EU_Grid_Voltage_Lower_Limit "Grid Voltage Lower Limit" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#grid_voltage_lower_limit" }
Number:ElectricPotential DeyeSUN2000G3EU_Grid_Voltage_Upp_Limit "Grid Voltage Upp Limit" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#grid_voltage_upp_limit" }
Number DeyeSUN2000G3EU_Island_Protection_Enable "Island Protection Enable" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#island_protection_enable" }
Switch DeyeSUN2000G3EU_ONOFF_Enable "ON-OFF Enable" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#onminusoff_enable" }
Number DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Percentage "Overfrequency And Load Reduction Percentage" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#overfrequency_and_load_reduction_percentage" }
Number:Frequency DeyeSUN2000G3EU_Overfrequency_And_Load_Reduction_Starting_Point "Overfrequency And Load Reduction Starting Point" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#overfrequency_and_load_reduction_starting_point" }
Switch DeyeSUN2000G3EU_OverfrequencyLoadshedding_Enable "Overfrequency&Load-shedding Enable" <energy> ["Point"] { channel="solarman:logger:4acf902443:grid#overfrequencyloadminusshedding_enable" }
Number DeyeSUN2000G3EU_AC_Version_Number "AC Version. Number" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#ac_version_number" }
Number DeyeSUN2000G3EU_Communication_Protocol_Version "Communication Protocol Version" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#communication_protocol_version" }
String DeyeSUN2000G3EU_DC_Master_Firmware_Version "DC Master Firmware Version" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#dc_master_firmware_version" }
String DeyeSUN2000G3EU_Hardware_Version "Hardware Version" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#hardware_version" }
Number DeyeSUN2000G3EU_Inverter_ID "Inverter ID" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#inverter_id" }
Switch DeyeSUN2000G3EU_Power_Factor_Regulation "Power Factor Regulation" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#power_factor_regulation" }
Number:Temperature DeyeSUN2000G3EU_Radiator_Temperature "Radiator Temperature" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#radiator_temperature" }
Number:Power DeyeSUN2000G3EU_Rated_Power "Rated Power" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#rated_power" }
Switch DeyeSUN2000G3EU_Restore_Factory_Settings "Restore Factory Settings" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#restore_factory_settings" }
Number DeyeSUN2000G3EU_Running_Status "Running Status" <status> ["Point"] { channel="solarman:logger:4acf902443:inverter#running_status" }
Switch DeyeSUN2000G3EU_Soft_Start_Enable "Soft Start Enable" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#soft_start_enable" }
Number DeyeSUN2000G3EU_Startup_Selfchecking_Time_ "Start-up Self-checking Time " <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#startminusup_selfminuschecking_time_" }
Number:Power DeyeSUN2000G3EU_Total_AC_Output_Power_Active "Total AC Output Power (Active)" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#total_ac_output_power_active" }
Number DeyeSUN2000G3EU_Update_Time "Update Time" <energy> ["Point"] { channel="solarman:logger:4acf902443:inverter#update_time" }
Switch DeyeSUN2000G3EU_Active_Power_Regulations "Active Power Regulations" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#active_power_regulations" }
Number:Energy DeyeSUN2000G3EU_Daily_Production "Daily Production" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#daily_production" }
Number:Energy DeyeSUN2000G3EU_Daily_Production_1 "Daily Production 1" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#daily_production_1" }
Number:Energy DeyeSUN2000G3EU_Daily_Production_2 "Daily Production 2" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#daily_production_2" }
Number:Energy DeyeSUN2000G3EU_Daily_Production_3 "Daily Production 3" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#daily_production_3" }
Number:Energy DeyeSUN2000G3EU_Daily_Production_4 "Daily Production 4" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#daily_production_4" }
Number:ElectricCurrent DeyeSUN2000G3EU_PV1_Current "PV1 Current" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv1_current" }
Number:ElectricPotential DeyeSUN2000G3EU_PV1_Voltage "PV1 Voltage" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv1_voltage" }
Number:ElectricCurrent DeyeSUN2000G3EU_PV2_Current "PV2 Current" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv2_current" }
Number:ElectricPotential DeyeSUN2000G3EU_PV2_Voltage "PV2 Voltage" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv2_voltage" }
Number:ElectricCurrent DeyeSUN2000G3EU_PV3_Current "PV3 Current" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv3_current" }
Number:ElectricPotential DeyeSUN2000G3EU_PV3_Voltage "PV3 Voltage" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv3_voltage" }
Number:ElectricCurrent DeyeSUN2000G3EU_PV4_Current "PV4 Current" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv4_current" }
Number:ElectricPotential DeyeSUN2000G3EU_PV4_Voltage "PV4 Voltage" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#pv4_voltage" }
Number:Energy DeyeSUN2000G3EU_Total_Production "Total Production" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#total_production" }
Number:Energy DeyeSUN2000G3EU_Total_Production_1 "Total Production 1" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#total_production_1" }
Number:Energy DeyeSUN2000G3EU_Total_Production_2 "Total Production 2" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#total_production_2" }
Number:Energy DeyeSUN2000G3EU_Total_Production_3 "Total Production 3" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#total_production_3" }
Number:Energy DeyeSUN2000G3EU_Total_Production_4 "Total Production 4" <energy> ["Point"] { channel="solarman:logger:4acf902443:solar#total_production_4" }
```

**Comment by catalinsanda** [comment]
You simply drop it in the `$OPENHAB_CONF/items` folder.

Remember to delete the UI defined items, otherwise it will probably complain of duplicates.

**Comment by catalinsanda** [comment]
Out of curiosity how did you export the items to a file? Using a script like this: https://community.openhab.org/t/converting-ui-items-back-to-items-file/117251/23 or does OpenHAB has some export/convert functionality?

Thanks,
Catalin

**Comment by bolausson** [comment]
Yes I used this script (Just realised that I missed adding the link to the script to "exporting")

Before I delete the UI defined items, I'll check tomorrow if "updating" them dis the trick or not. If not, I will delete them and drop the file in the dir you mentioned.

**Comment by bolausson** [comment]
Didn't help - only "one set of metrics" is updating :(
I will remove the items, links, binding etc and start with the items.file

At least it looks like it is installed only once.

```
openhab> bundle:list                                                                                                                                                                                    
START LEVEL 100 , List Threshold: 50
 ID │ State  │ Lvl │ Version                │ Name
────┼────────┼─────┼────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 31 │ Active │  80 │ 2.14.1                 │ Jackson-annotations
 32 │ Active │  80 │ 2.14.1                 │ Jackson-core
 33 │ Active │  80 │ 2.14.1                 │ jackson-databind
 34 │ Active │  80 │ 2.14.1                 │ Jackson-dataformat-YAML
 35 │ Active │  80 │ 2.14.1                 │ Jackson datatype: JSR310
 36 │ Active │  80 │ 6.2.6                  │ Woodstox
 37 │ Active │  80 │ 2.9.1                  │ Gson
 38 │ Active │  80 │ 30.1.0.jre             │ Guava: Google Core Libraries for Java
 39 │ Active │  80 │ 1.0.1                  │ Guava InternalFutureFailureAccess and InternalFutures
 40 │ Active │  80 │ 5.0.1.v20210324-2015   │ Google Guice - Core Library
 41 │ Active │  80 │ 5.12.1                 │ jna
 42 │ Active │  80 │ 5.12.1                 │ jna-platform
 44 │ Active │  80 │ 4.8.149.v20220915-0556 │ ClassGraph
 45 │ Active │  80 │ 0.17.1                 │ Directory-Watcher
 46 │ Active │  80 │ 2.1.9                  │ swagger-annotations
 47 │ Active │  80 │ 2.1.9                  │ swagger-core
 48 │ Active │  80 │ 2.1.9                  │ swagger-integration
 49 │ Active │  80 │ 2.1.9                  │ swagger-jaxrs2
 50 │ Active │  80 │ 2.1.9                  │ swagger-models
 52 │ Active │  80 │ 2.0.2                  │ Jakarta Bean Validation API
 54 │ Active │  80 │ 2.3.3                  │ Jakarta XML Web Services API
 55 │ Active │  80 │ 1.2                    │ javax.annotation API
 56 │ Active │  80 │ 3.5.8                  │ JmDNS
 57 │ Active │  80 │ 2.1.3                  │ Units of Measurement API
 59 │ Active │  80 │ 0.5.10                 │ Jollyday
 60 │ Active │  80 │ 3.2.0.v201101311130    │ ANTLR Runtime
 61 │ Active │  80 │ 1.0.0.v201105210816    │ Aopalliance Plug-in
 62 │ Active │  80 │ 1.2.2                  │ org.apache.aries.component-dsl.component-dsl
 63 │ Active │  80 │ 1.0.1                  │ Apache Aries JAX-RS Specification API
 64 │ Active │  80 │ 2.0.0                  │ Apache Aries JAX-RS Whiteboard
 65 │ Active │  80 │ 1.3.6                  │ Apache Aries SPI Fly Dynamic Weaving Bundle
 66 │ Active │  80 │ 3.12.0                 │ Apache Commons Lang
 67 │ Active │  80 │ 3.4.5                  │ Apache CXF Core
 68 │ Active │  80 │ 3.4.5                  │ Apache CXF Runtime JAX-RS Frontend
 69 │ Active │  80 │ 3.4.5                  │ Apache CXF JAX-RS Client
 70 │ Active │  80 │ 3.4.5                  │ Apache CXF JAX-RS Server-Side Events Support
 71 │ Active │  80 │ 3.4.5                  │ Apache CXF Runtime Security functionality
 72 │ Active │  80 │ 3.4.5                  │ Apache CXF Runtime HTTP Transport
 78 │ Active │  80 │ 4.4.3                  │ Apache Karaf :: OSGi Services :: Event
 89 │ Active │  80 │ 1.0.0.2                │ Apache ServiceMix :: Bundles :: javax.inject
 96 │ Active │  80 │ 2.2.5                  │ XmlSchema Core
 97 │ Active │  80 │ 0.9.3                  │ jose4j
 98 │ Active │  80 │ 2.17.0.v20190920-0401  │ EMF Common
 99 │ Active │  80 │ 2.20.0.v20190920-0401  │ EMF Ecore
100 │ Active │  80 │ 2.14.0.v20190528-0725  │ EMF Change Model
101 │ Active │  80 │ 2.16.0.v20190528-0725  │ EMF XML/XMI Persistence
102 │ Active │  80 │ 3.13.0.v20200828-1034  │ Common Eclipse Runtime
104 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: ALPN :: Client
105 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: ALPN :: JDK9 Client Implementation
111 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: HTTP2 :: Client
118 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Proxy
126 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Websocket :: API
127 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Websocket :: Client
128 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Websocket :: Common
129 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Websocket :: Server
130 │ Active │  80 │ 9.4.50.v20221201       │ Jetty :: Websocket :: Servlet Interface
132 │ Active │  80 │ 0.19.0.v20221118-0359  │ org.eclipse.lsp4j
133 │ Active │  80 │ 0.19.0.v20221118-0359  │ org.eclipse.lsp4j.jsonrpc
134 │ Active │  80 │ 2.29.0.v20221121-0915  │ Xtend Runtime Library
135 │ Active │  80 │ 2.29.0.v20221121-0915  │ Xtend Macro Interfaces
136 │ Active │  80 │ 2.29.0.v20221121-0917  │ Xtext
137 │ Active │  80 │ 2.29.0.v20221121-0924  │ Xtext Common Types
138 │ Active │  80 │ 2.29.0.v20221121-0917  │ Xtext IDE Core
139 │ Active │  80 │ 2.29.0.v20221121-0917  │ Xtext Utility
140 │ Active │  80 │ 2.29.0.v20221121-0924  │ Xbase Model
141 │ Active │  80 │ 2.29.0.v20221121-0924  │ Xbase Generic IDE Services
142 │ Active │  80 │ 2.29.0.v20221121-0915  │ Xbase Runtime Library
146 │ Active │  80 │ 9.4.0                  │ org.objectweb.asm.commons
147 │ Active │  80 │ 9.4.0                  │ org.objectweb.asm.tree
148 │ Active │  80 │ 9.4.0                  │ org.objectweb.asm.tree.analysis
149 │ Active │  80 │ 9.4.0                  │ org.objectweb.asm.util
150 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Core
151 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Add-on XML
152 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Marketplace Add-on Services
153 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Community Marketplace Add-on Service :: Karaf
154 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Audio
155 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: JAAS Authentication
156 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: OAuth2Client
157 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Automation
158 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Automation Media Modules
159 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Automation Script Modules
160 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Automation Script RuleSupport
161 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Automation REST API
162 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration Core
163 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration Discovery
164 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration mDNS Discovery
165 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration Dispatcher
166 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Ephemeris
167 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: ID
168 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Console
169 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Console for OSGi runtime Karaf
170 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: HTTP Interface
171 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: HTTP Interface Authentication
173 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Monitor
174 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Network I/O
175 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: REST Interface
176 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Audio REST Interface
177 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Authentication Support for the REST Interface
178 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: REST Interface :: Core
179 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: REST mDNS Announcer
180 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Sitemap REST Interface
181 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: SSE Interface
182 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: REST Interface :: Swagger 1
183 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Transformation REST Interface
184 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: UI REST Interface
185 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Voice REST Interface
186 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: mDNS Service
187 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: WebSocket
188 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Karaf Integration
189 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Core
190 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Items
191 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Item IDE
192 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Items Runtime
193 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Language Server
194 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Persistence
195 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Persistence IDE
196 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Persistence Runtime
197 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Rules
198 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Rule IDE
199 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Rules Runtime
200 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Script
201 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Script IDE
202 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Script Runtime
203 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Sitemap
204 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Sitemap IDE
205 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Sitemap Runtime
206 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Thing
207 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Thing IDE
208 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Model Thing Runtime
209 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Persistence
210 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Semantics
211 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: JSON Storage
212 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Thing
213 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Transformation Service
214 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: UI
215 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: UI Icon Support
216 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Voice
217 │ Active │  80 │ 4.0.1                  │ openHAB UI :: Bundles :: Main UI
218 │ Active │  80 │ 4.0.1                  │ openHAB UI :: Bundles :: Icon Set :: Classic
228 │ Active │  80 │ 1.0.0.201802012106     │ org.osgi:org.osgi.service.jaxrs
229 │ Active │  80 │ 1.5.0                  │ ThreeTen-Extra
230 │ Active │  80 │ 1.33.0                 │ SnakeYAML
231 │ Active │  80 │ 2.1                    │ SI Units
232 │ Active │  80 │ 2.1.0                  │ SI Quantities
233 │ Active │  80 │ 4.2.1                  │ Stax2 API
234 │ Active │  80 │ 2.1.2                  │ Units of Measurement Reference Implementation
235 │ Active │  80 │ 2.1.0                  │ Units of Measurement Common Library for Java
236 │ Active │  80 │ 1.4.20                 │ XStream Core
237 │ Active │  80 │ 5.2.1.OH1              │ nrjavaserial
238 │ Active │  80 │ 3.9.0                  │ Apache Commons Net
239 │ Active │  80 │ 2.7.0                  │ JUPnP Library
240 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: deCONZ Binding
241 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Homematic Binding
242 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: hue Binding
243 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: MELCloud Binding
244 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Pushover Binding
245 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: UniFi Binding
246 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: ZWave Binding
247 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration UPnP Discovery
248 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration USB-Serial Discovery
249 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration USB-Serial Discovery for Linux using sysfs scanning
250 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration USB-Serial Discovery using ser2net mDNS scanning
251 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Configuration Serial
252 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Serial Transport
253 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Serial Transport for RXTX
254 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: Serial Transport for RFC2217
255 │ Active │  80 │ 4.0.1                  │ openHAB Core :: Bundles :: UPnP Transport
256 │ Active │  80 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Persistence Service :: InfluxDB
257 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: Exec
258 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: JSonPath
259 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: Map
260 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: RegEx
261 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: Scale
262 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: XPath
263 │ Active │  75 │ 4.0.1                  │ openHAB Add-ons :: Bundles :: Transformation Service :: Xslt
264 │ Active │  80 │ 4.0.1                  │ openHAB UI :: Bundles :: Basic UI
265 │ Active │  80 │ 4.0.1                  │ openHAB UI :: Bundles :: HABPanel UI
267 │ Active │  80 │ 0.2.2                  │ openHAB Add-ons :: Bundles :: Solarman Binding
```

**Comment by bolausson** [comment]
Okay, it works - I remove all itmes, thing, links & removed the binding.
Restarted added the binding, added the thing (with same UID) and imported the above item file via the UI.

No clue what went wrong in the first place but now it is fine.

The only thing that doesn't seem to return anything is the "Update time" channel.

Thanks for all your support,
Bjoern

**Comment by catalinsanda** [comment]
Glad it mostly works, regarding the "Update time" is to be expected, I haven't implemented parsing those types of values (rules 7,8,9 from the inverter definition file), because they were not present in the definition that I use, so there was no way to test it.

I'll take a stab at it in the next few days and let you know. Keeping this open in the meantime.

Regards,
Catalin

**Comment by catalinsanda** [comment]
Hey,

Can you try the latest release: https://github.com/catalinsanda/org.openhab.binding.solarman/releases/tag/v0.2.4 ?
Hopefully it will handle the update time as well. I've implemented the support based on the hex dumps you provided in another issue (this was fun), so I'm not sure if it will work like it's supposed to. 

Please keep in mind that the value will be datetime so it should be mapped to a DateTime item.

Let me know if it works.

Thanks,
Catalin

**Comment by bolausson** [comment]
Sure, I'll give it a try later today!
:-)

**Comment by bolausson** [comment]
> Hey,
> 
> Can you try the latest release: https://github.com/catalinsanda/org.openhab.binding.solarman/releases/tag/v0.2.4 ? Hopefully it will handle the update time as well. I've implemented the support based on the hex dumps you provided in another issue (this was fun), so I'm not sure if it will work like it's supposed to.
> 
> Please keep in mind that the value will be datetime so it should be mapped to a DateTime item.
> 
> Let me know if it works.
> 
> Thanks, Catalin

It works:

![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/10600678/16ab0b99-63ed-4ca6-829e-881862b4a49d)
![image](https://github.com/catalinsanda/org.openhab.binding.solarman/assets/10600678/fad51450-b8db-4ab8-9fcd-656f85a83f2b)

Thanks a lot for all the addional values and help you provided!

**Comment by catalinsanda** [comment]
Glad it works, and nice graphs, btw!

Closing this.
